### PR TITLE
Optimize building node infos by using map with pods for nodes.

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**
Optimize building node infos by using map with pods for nodes.

### CODE REVIEW

File: cluster-autoscaler/simulator/nodes.go [hunk 1]
```diff
@@ -23,25 +23,14 @@ import (
...
-		return []*apiv1.Pod{}, errors.ToAutoscalerError(errors.ApiCallError, err)
-	}
-	allPods := make([]*apiv1.Pod, 0)
-	for _, p := range pods {
-		if p.Spec.NodeName == nodename {
-			allPods = append(allPods, p)
-		}
-	}
-
+func GetRequiredPodsForNode(nodename string, podsForNodes map[string][]*apiv1.Pod) ([]*apiv1.Pod, errors.AutoscalerError) {
```
Comment (losipiuk) [inline cluster-autoscaler/simulator/nodes.go]: btw. Does this have to be public.

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/simulator/nodes.go]: Done.

**Pull request merged**